### DISCUSSION TITLE
windows更新后，canvas一直报skia.win32-x64-msvc.node，打开不了

### DISCUSSION THREAD

**Discussion by Modilay** [Tsuk1ko/CQ-picfinder-robot#282] [问题求助]
![image](https://user-images.githubusercontent.com/61842831/148219455-6721f261-b457-4214-b36d-4634332a2a53.png)

按照另一个帖子安装了windows-build-tools还是一样

**Comment by Tsuk1ko** [comment]
尝试删掉 node_modules 文件夹重新安装一次依赖

**Reply by Modilay** [reply]
> 尝试删掉 node_modules 文件夹重新安装一次依赖

_(:з)∠)_这个也试过了，node_modules删了重新npm i后还是一样，然后重装了下node后也还是一样

**Reply by Tsuk1ko** [reply]
node 版本是？

**Reply by Tsuk1ko** [reply]
另外 npm i 的过程中有什么错误日志吗

以及到你截图里报错找不到的那个 .node 文件的路径看看文件是不是真的不存在

**Reply by Modilay** [reply]
> 另外 npm i 的过程中有什么错误日志吗
> 
> 以及到你截图里报错找不到的那个 .node 文件的路径看看文件是不是真的不存在

![image](https://user-images.githubusercontent.com/61842831/148317813-efaf4f0e-b520-44b8-aca2-5a8a138829d6.png)

node是刚从官网下的新的 16.13.1，npm i的时候，没有错误.👇是npm i的结果

![image](https://user-images.githubusercontent.com/61842831/148318229-8d32c67e-a07a-4561-bbcb-1561243983ba.png)

.node文件时存在的
![image](https://user-images.githubusercontent.com/61842831/148318327-084b43c0-faab-473a-867d-371f7f87f72f.png)

系统是windows server 2016

**Reply by Tsuk1ko** [reply]
试试把项目根目录下的 package.json 里的 @napi-rs/canvas 版本改成固定 0.1.15
```
"@napi-rs/canvas": "0.1.15",
```
然后重新安装下依赖试试

**Reply by Modilay** [reply]
> 试试把项目根目录下的 package.json 里的 @napi-rs/canvas 版本改成固定 0.1.15
> 
> ```
> "@napi-rs/canvas": "0.1.15",
> ```
> 
> 然后重新安装下依赖试试

![image](https://user-images.githubusercontent.com/61842831/148334233-9798f321-f093-447e-9e17-dbfe55d61348.png)


换0.1.15后变酱紫了

**Reply by Tsuk1ko** [reply]
那怪了，以前你是怎么跑起来的……

**Reply by Modilay** [reply]
> 那怪了，以前你是怎么跑起来的……

以前倒是没事，只不过昨天发现进程没了，应该没了好几天了，然后昨天就去更新了下，然后就一直这样打不开了。怪了

**Reply by Tsuk1ko** [reply]
等晚些时候我弄一个兼容方案吧。。

**Reply by Modilay** [reply]
> 等晚些时候我弄一个兼容方案吧。。

嗯，谢了，幸苦了

**Comment by Tsuk1ko** [comment]
已经发布了 v2.32.0，可以切换 canvas 库，详见 [wiki](https://github.com/Tsuk1ko/cq-picsearcher-bot/wiki/%E9%85%8D%E7%BD%AE%E6%96%87%E4%BB%B6%E8%AF%B4%E6%98%8E#%E8%87%AA%E5%AE%9A%E4%B9%89-canvas-%E5%BA%93)
